src/components/content/order/ServiceList.tsx

```diff
@@ -0,0 +1,210 @@
+            sorter: (service1, service2) => sortVersion(service1, service2),
+        },
+        {
+            title: 'Csp',
+            dataIndex: 'csp',
+            filters: cspFilters,
+            filterMode: 'tree',
+            filterSearch: true,
+            onFilter: (value: string | number | boolean, record) => record.csp.startsWith(value.toString()),
+        },
+        {
+            title: 'Flavor',
+            dataIndex: 'flavor',
+        },
+        {
+            title: 'ServiceState',
+            dataIndex: 'serviceState',
+        },
+        {
+            title: 'Operation',
+            dataIndex: 'operation',
+            render: (text, record) => (
+                <Space size='middle'>
+                    <Button type='primary' icon={<CopyOutlined />}>
+                        migrate
+                    </Button>
+                    <Button type='primary' icon={<CloseCircleOutlined />}>
+                        destroy
+                    </Button>
+                </Space>
+            ),
+        },
+    ];
+
+    function getCspFilters(resp: ServiceVo[]) {
+        let filters: ColumnFilterItem[] = [];
+        let cspSet = new Set<string>('');
+        resp.forEach((v, index) => {
+            cspSet.add(v.csp);
+        });
+        cspSet.forEach((csp) => {
+            let filter = {
+                text: csp,
+                value: csp,
+            };
+            filters.push(filter);
+        });
+        setCspFilters(filters);
+    }
+    function getCategoryFilters(resp: ServiceVo[]) {
+        let filters: ColumnFilterItem[] = [];
+        let categorySet = new Set<string>('');
+        resp.forEach((v, index) => {
+            categorySet.add(v.category);
+        });
+        categorySet.forEach((category) => {
+            let filter = {
+                text: category,
+                value: category,
+            };
+            filters.push(filter);
+        });
+        setCategoryFilters(filters);
+    }
+
+    function getVersionFilters(resp: ServiceVo[]) {
+        let filters: ColumnFilterItem[] = [];
+        let versionSet = new Set<string>('');
+        resp.forEach((v, index) => {
+            versionSet.add(v.version);
+        });
+        versionSet.forEach((version) => {
+            let filter = {
+                text: version,
+                value: version,
+            };
+            filters.push(filter);
+        });
+        setVersionFilters(filters);
+    }
+    function getNameFilters(resp: ServiceVo[]) {
+        let filters: ColumnFilterItem[] = [];
+        let nameSet = new Set<string>('');
+        resp.forEach((v, index) => {
+            nameSet.add(v.name);
+        });
+        nameSet.forEach((name) => {
+            let filter = {
+                text: name,
+                value: name,
+            };
+            filters.push(filter);
+        });
+        setNameFilters(filters);
+    }
+    const getServices = useCallback(() => {
```

Comment:
can this not be a simple function instead of useCallback? because the dependency is empty which means the function is always the same.

Reply:
fixed.